src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -274,7 +277,54 @@ private ValueTask<HttpConnection> GetConnectionAsync(HttpRequestMessage request,
             // TODO: For now, we don't support proxy authentication in this scenario.
             // This will get fixed when we refactor proxy auth handling.
 
-            HttpResponseMessage tunnelResponse = await _poolManager.SendAsync(tunnelRequest, null, cancellationToken).ConfigureAwait(false);
+            // TODO: shall we honor _preAuthenticate and send Basic auth when creds are available?
+            HttpResponseMessage tunnelResponse = await _poolManager.SendAsync(tunnelRequest, null, null, cancellationToken).ConfigureAwait(false);
```

Comment:
We shouldn't just copy and paste this code.  We should refactor the existing code to be usable in both scenarios. 

The existing code relies on a couple settings, plus a way to resend requests -- that's the innerHandler stuff.  We've used HttpMessageHandler to implement these "inner handlers" but we could change if we wanted to.  

So to reuse the code, we need to be able to pass it an inner handler that will call _poolManger.SendAsync to resend the request as appropriate.  And we need to tweak the code that handles settings so that they can be passed in either from here or from the other usage.

I think that should work.  Let me know if I can assist.